rtl/include/eclhelper.hpp

```diff
@@ -2973,14 +3277,53 @@ struct IGlobalCodeContext
     virtual void doNotify(char const * name, char const * text, const char * target) = 0;
 };
 
+// Wraps an IGlobalCodeContext so that ::queryCodeContext returns
+// an ICodeContext which returns the workflow id.
+struct GlobalCodeContextExtra : implements IGlobalCodeContext
+{
+private:
+    // Wraps ICodeContext and overrides getWorkflowId() method
+    class CodeContextEx : public IndirectCodeContext
+    {
+        unsigned wfid;
+    public:
+        CodeContextEx(ICodeContext * _ctx, unsigned _wfid) : IndirectCodeContext(_ctx), wfid(_wfid) {}
+        virtual ~CodeContextEx() {};
+        virtual unsigned getWorkflowId() const override { return wfid; }
+    };
+    IGlobalCodeContext * gctx;
+    CodeContextEx * codeContextEx;
```

Comment:
Minor.  This could be a member object rather than a pointer, avoiding the need for the new/delete.